app/src/main/java/com/android/sample/ui/swipePage/SwipePage.kt

```diff
@@ -389,33 +404,119 @@ fun RecipeDisplay(
+                      displayLike = animationTarget == END_ANIMATION
+                      displayDisLike = animationTarget == -END_ANIMATION
+                    },
+                    blockRetrieveNextRecipe = { retrieveNextRecipe = false })
+              }
+              LaunchedEffect(offsetXCard2.value) {
+                animateCard(
+                    offsetX = offsetXCard2,
+                    screenWidth = screenWidth,
+                    isClicking = isClicking,
+                    isSwiping = isSwiping,
+                    retrieveNextRecipe = retrieveNextRecipe,
+                    recipesViewModel = recipesViewModel,
+                    startSwiping = { isSwiping = true },
+                    endSwiping = { isSwiping = false },
+                    updateDisplayCard = { displayCard1 = true },
+                    displayIcons = { animationTarget ->
+                      displayLike = animationTarget == END_ANIMATION
+                      displayDisLike = animationTarget == -END_ANIMATION
+                    },
+                    blockRetrieveNextRecipe = { retrieveNextRecipe = true })
               }
             }
       }
 }
 
+private fun manageRecipeLiked(
+    offsetX: Animatable<Float, AnimationVector1D>,
+    retrieveRecipeAuthorized: Boolean,
+    swipeThreshold: Float,
+    currentRecipe: Recipe?,
+    userViewModel: UserViewModel,
+    authorizeRetrieveNextRecipe: () -> Unit
+) {
+  if (abs(offsetX.value) > swipeThreshold) {
+    authorizeRetrieveNextRecipe()
+    if (retrieveRecipeAuthorized && MIN_OFFSET_X < offsetX.value && currentRecipe != null) {
+      userViewModel.addRecipeToUserLikedRecipes(currentRecipe)
+    }
+  }
+}
+
+/**
+ * Animate the Swipe of the recipe card
+ *
+ * @param offsetX - Offset for the swipe animation
+ * @param screenWidth - Screen width
+ * @param isClicking - Clicking state
+ * @param isSwiping - Swiping state
+ * @param retrieveNextRecipe - Retrieve next recipe state
+ * @param recipesViewModel - Recipes View Model
+ * @param startSwiping - Start Swiping function
+ * @param endSwiping - End Swiping function
+ * @param updateDisplayCard - Update Display Card function
+ * @param displayIcons - Display Icons function
+ * @param blockRetrieveNextRecipe - Block Retrieve Next Recipe function
+ */
+private suspend fun animateCard(
+    offsetX: Animatable<Float, AnimationVector1D>,
+    screenWidth: Float,
+    isClicking: Boolean,
+    isSwiping: Boolean,
+    retrieveNextRecipe: Boolean,
+    recipesViewModel: RecipesViewModel,
+    startSwiping: () -> Unit,
+    endSwiping: () -> Unit,
+    updateDisplayCard: () -> Unit,
+    displayIcons: (Float) -> Unit,
+    blockRetrieveNextRecipe: () -> Unit
+) {
+  val swipeThreshold = screenWidth * SWIPE_THRESHOLD
+
+  if (abs(offsetX.value) > screenWidth) {
+    startSwiping()
+    delay(ANIMATION_SWIPE_TIME.toLong())
+    updateDisplayCard()
+    endSwiping()
+    offsetX.snapTo(INITIAL_OFFSET_X)
+  }
```

Comment:
Could you please add some internal comments to this function? To be honest, I’m finding it quite challenging to understand what’s going on. A bit of additional context would really help clarify the logic. Thank you!

Reply:
Yes, of course! It's all set now.

Reply:
Thanks LGTM now!